marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs

```diff
@@ -3,142 +3,236 @@
 
 module Spec.Marconi.ChainIndex.Indexers.Utxo.UtxoIndex (tests) where
 
-import Cardano.Api qualified as C
-import Control.Lens (filtered, folded, toListOf)
+import Control.Lens (each, filtered, folded, toListOf)
+import Control.Lens.Operators ((%~), (&), (^.))
 import Control.Monad (forM, forM_, void)
 import Control.Monad.IO.Class (liftIO)
 import Data.Aeson qualified as Aeson
+import Data.Functor ((<&>))
 import Data.List qualified as List
 import Data.List.NonEmpty (nonEmpty)
-import Data.Maybe (isJust, isNothing, mapMaybe)
+import Data.Map (Map)
+import Data.Map qualified
+import Data.Maybe (fromMaybe, isJust, isNothing, mapMaybe)
 import Data.Set (Set)
 import Data.Set qualified as Set
-import Gen.Marconi.ChainIndex.Indexers.Utxo (genEventWithShelleyAddressAtChainPoint, genShelleyEraUtxoEvents,
-                                             genUtxoEvents)
+
+import Cardano.Api qualified as C
+import Gen.Marconi.ChainIndex.Indexers.Utxo (genShelleyEraUtxoEvents, genUtxoEvents)
 import Gen.Marconi.ChainIndex.Indexers.Utxo qualified as UtxoGen
-import Gen.Marconi.ChainIndex.Mockchain (mockBlockTxs)
+import Gen.Marconi.ChainIndex.Mockchain (MockBlock (mockBlockChainPoint, mockBlockTxs))
 import Gen.Marconi.ChainIndex.Types (genChainPoints)
-import Hedgehog (Property, cover, forAll, property, (===))
-import Hedgehog qualified
-import Hedgehog.Gen qualified as Gen
-import Hedgehog.Range qualified as Range
 import Helpers (addressAnyToShelley)
 import Marconi.ChainIndex.Indexers.Utxo (StorableEvent (ueInputs, ueUtxos))
 import Marconi.ChainIndex.Indexers.Utxo qualified as Utxo
 import Marconi.ChainIndex.Types (TargetAddresses)
 import Marconi.Core.Storable (StorableQuery)
 import Marconi.Core.Storable qualified as Storable
+
+import Hedgehog (Gen, Property, cover, forAll, property, (===))
+import Hedgehog qualified
+import Hedgehog.Gen qualified as Gen
+import Hedgehog.Range qualified as Range
 import Test.Tasty (TestTree, testGroup)
 import Test.Tasty.Hedgehog (testPropertyNamed)
 
 tests :: TestTree
 tests = testGroup "Spec.Marconi.ChainIndex.Indexers.Utxo"
-    [ testPropertyNamed
-        "filter UtxoEvent for Utxos with address in the TargetAddress"
-        "eventsAtAddressTest"
-        eventsAtAddressTest
+      [ testPropertyNamed
+        "marconi-chain-index-utxo regression-test, store and retrieve UtxoEvents by address, then test that there are no `Spent` in the retrieved Utxo(s)"
```

Comment:
No need to say it's a regression test.

Also, no need to put "marconi-chain-index-utxo" for everything. The `testGroup` already mentions what module we're testing.

How about a test description like: "All queried UTXOs by address should be unspent.